extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/ColorThingHandler.java

```diff
@@ -0,0 +1,300 @@
+import java.util.stream.IntStream;
+
+import org.eclipse.smarthome.binding.dmx.DmxBindingConstants.ListenerType;
+import org.eclipse.smarthome.binding.dmx.internal.DmxBridgeHandler;
+import org.eclipse.smarthome.binding.dmx.internal.DmxThingHandler;
+import org.eclipse.smarthome.binding.dmx.internal.Util;
+import org.eclipse.smarthome.binding.dmx.internal.ValueSet;
+import org.eclipse.smarthome.binding.dmx.internal.action.FadeAction;
+import org.eclipse.smarthome.binding.dmx.internal.multiverse.BaseDmxChannel;
+import org.eclipse.smarthome.binding.dmx.internal.multiverse.DmxChannel;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ColorThingHandler} is responsible for handling commands, which are
+ * sent to the dimmer.
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+
+public class ColorThingHandler extends DmxThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_COLOR);
+
+    private final Logger logger = LoggerFactory.getLogger(ColorThingHandler.class);
+
+    private final List<DmxChannel> channels = new ArrayList<DmxChannel>();
+
+    private final List<Integer> currentValues = new ArrayList<Integer>();
+    private HSBType currentColor = new HSBType();
+
+    private ValueSet turnOnValue = new ValueSet(0, -1, DmxChannel.MAX_VALUE);
+    private ValueSet turnOffValue = new ValueSet(0, -1, DmxChannel.MIN_VALUE);
+
+    private int fadeTime = 0, dimTime = 0;
+
+    private boolean isDimming = false;
+
+    public ColorThingHandler(Thing dimmerThing) {
+        super(dimmerThing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("received command {} in channel {}", command, channelUID);
+        ValueSet targetValueSet = new ValueSet(fadeTime, -1);
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS_R:
+                if (command instanceof RefreshType) {
+                    logger.trace("sending update on refresh to channel {}:brightness_r", this.thing.getUID());
+                    updateState(channelUID, Util.toPercentValue(currentValues.get(0)));
```

Comment:
Maybe there is a misunderstanding on my side regarding the `Refresh` command. What exactly is supposed to happen? From my understanding, the bidning should update the channel state so that it reflects the device state. Is that wrong? 
Regarding the `currentValue`: There is no real status feedback as DMX is one-way communication. The binding keeps track of the (assumed) value of each DMX channel in the universe. So I might get that information from the universe (which is the same as in `currentValue` if the binding is working as it should).

Reply:
Ah, Ok now I see. So the misunderstanding was on my side. You are right, the `refresh` command is issued by the framework to query the current state of the real physical device. So if you only have a one way communication, returning the one from your universe is the right thing to do.